armi/reactor/tests/test_assemblies.py

```diff
@@ -1137,6 +1137,44 @@ def test_carestianCoordinates(self):
         )
         self.assertEqual(a.coords(), (2.0, 2.0))
 
+    def test_pinPlenumVolume(self):
+        """
+        Test the volume of a pin in the assembly's plenum.
+        """
+        assembly = assemblies.Assembly("dummy")
+        self.assertEqual(0.0, assembly.getPinPlenumVolumeInCubicMeters())
+
+        pinPlenumVolume = 7.853981633974482e-06
+        testAssemblies = buildTestAssemblies()
+        plenumBlock = makeTestAssembly(
+            1, 2, grids.CartesianGrid.fromRectangle(2.0, 2.0))
+        plenumBlock.setType("plenum", Flags.PLENUM)
+        self.Assembly.setBlockMesh([10.0, 20.0, 30.0], conserveMassFlag="auto")
+        plenumBlock.append(testAssemblies[0])
+        self.Assembly.append(plenumBlock)
+
+        self.assertEqual(
+            pinPlenumVolume, self.Assembly.getPinPlenumVolumeInCubicMeters())
+
+    def test_averagePlenumTemperature(self):
+        """
+        Test an assembly's average plenum temperature with a single block outlet.
+        """
+        averagePlenumTemp = 42.0
+        tempParam = (
+            assemblies.assemblyParameters.getAssemblyParameterDefinitions().__getitem__(
+                'THcoolantOutletT'
+                )
+            )
+        plenumBlock = makeTestAssembly(
+            1, 2, grids.CartesianGrid.fromRectangle(1.0, 1.0))
+        plenumBlock.setType("plenum", Flags.PLENUM)
+        plenumBlock.p[tempParam.name] = averagePlenumTemp
```

Comment:
Even more magical, `ParameterCollection`s have a fancy `__getattr__` implementation, so that you can do `plenumBlock.p.THcoolantOutletT`

Reply:
Thank you, I'm still internalizing Python's special methods.  It will be updated in the next commit.